app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt

```diff
@@ -31,25 +44,34 @@ class MosaicRepository(application: Application) {
             null,
             HttpMethod.GET
         ) { response ->
-            val graphResponse: JSONObject = response.jsonObject
-            val data: JSONArray = graphResponse.getJSONArray("data")
-            for (id in 0 until data.length()) {
-                val imageId: String = data.getJSONObject(id).getString("id")
-                GraphRequest(
-                    token,
-                    "/$imageId?fields=media_url",
-                    null,
-                    HttpMethod.GET
-                ) { response2 ->
-                    val graphResponse2: JSONObject = response2.jsonObject
-                    val imageUrl: String = graphResponse2.getString("media_url")
-                    loadImageByCoil(application, imageUrl)
-                }.executeAsync()
-            }
+            findImagesByIds(response, application, token)
         }.executeAsync()
     }
 
-    private fun loadImageByCoil(application: Application, imageUrl: String) {
+    private fun findImagesByIds(
+        response: GraphResponse,
+        application: Application,
+        token: AccessToken
+    ) {
+        val graphResponse: JSONObject = response.jsonObject
+        val data: JSONArray = graphResponse.getJSONArray("data")
+        for (id in 0 until data.length()) {
+            val imageId: String = data.getJSONObject(id).getString("id")
+            graphRequest = GraphRequest(
+                token,
+                "/$imageId?fields=media_url",
+                null,
+                HttpMethod.GET
+            ) { response2 ->
+                loadImageByCoil(response2, application)
+            }
+            graphRequest.executeAsync()
+        }
+    }
+
+    private fun loadImageByCoil(response2: GraphResponse, application: Application) {
+        val graphResponse2: JSONObject = response2.jsonObject
```

Comment:
Please rename all variables like graphResponse2, response2, response, etc. to something more readable, like "findImagesByIdsRequest", "findImagesByIdsResponse", etc.

Reply:
done